streams/shared/src/main/scala/zio/stream/StreamEffect.scala

```diff
@@ -365,12 +365,10 @@ private[stream] object StreamEffect extends Serializable {
       }
     }
 
-  final def fromIterator[R, E, A](iterator: ZManaged[R, E, Iterator[A]]): StreamEffect[R, E, A] =
+  final def fromIterator[R, E, A](iterator: Iterator[A]): StreamEffect[R, E, A] =
```

Comment:
How can we make this to work for #2334?

Reply:
I think we can't do this on StreamEffect right now, as it requires `unsafeRun`.

Reply:
Ok, then let's simplify the function signature. `R = Any, E = Nothing`?

Reply:
Oh yeah. Good catch.